test/framework/src/main/java/org/elasticsearch/test/ESSingleNodeTestCase.java

```diff
@@ -460,4 +463,11 @@ protected void ensureNoInitializingShards() {
     protected boolean enableConcurrentSearch() {
         return true;
     }
+
+    protected void awaitIndexShardCloseAsyncTasks() {
+        // ES-8334 TODO build this wait into the relevant APIs (especially, delete-index and close-index)
```

Comment:
I wonder about this, it seems that responding early is fine, given that for instance delete-index still does not guarantee files being deleted?

Reply:
Yeah I'm not sure about this either, you're right that we don't have particularly well-defined guarantees in this area today. We do rely on prompt deletion/closure in some tests tho, enough that it seemed worth a conversation. Again, not intending to solve this here, will follow up.